File: Android/app/src/main/java/com/dreamproject/inwords/presentation/viewScenario/octoGame/gameLevel/GameLevelViewModelFactory.java
package com.dreamproject.inwords.presentation.viewScenario.octoGame.gameLevel;

import com.dreamproject.inwords.data.repository.GameRepository;

import javax.inject.Inject;
import javax.inject.Singleton;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

@Singleton
public class GameLevelViewModelFactory implements ViewModelProvider.Factory {
    private final GameRepository gameRepository;

    @Inject
    GameLevelViewModelFactory(final GameRepository gameRepository) {
        this.gameRepository = gameRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(GameLevelViewModel.class)) {
            //noinspection unchecked
            return (T) new GameLevelViewModel(gameRepository);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }

}

File: Android/app/src/main/java/com/dreamproject/inwords/presentation/viewScenario/octoGame/gameLevel/GameLevelViewModelFactory.kt
package com.dreamproject.inwords.presentation.viewScenario.octoGame.gameLevel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.dreamproject.inwords.domain.interactor.game.GameInteractor
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class GameLevelViewModelFactory @Inject
internal constructor(private val gameInteractor: GameInteractor) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(GameLevelViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return GameLevelViewModel(gameInteractor) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}
